server/src/main/java/org/elasticsearch/search/suggest/completion/context/GeoContextMapping.java

```diff
@@ -314,7 +314,7 @@ public void validateReferences(Version indexVersionCreated, Function<String, Map
     public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
-        if (!super.equals(o)) return false;
```

Comment:
Do we have plans to enforce the idea templates for generating these equals methods?

Reply:
Not right now - do you happen to know whether this is possible?

Reply:
I would expect it is somehow possible to enforce via the idea-ext plugin (https://github.com/JetBrains/gradle-idea-ext-plugin/wiki) and defining a custom template. But not for sure.

Reply:
So actually IntelliJ seems to prefer using `Object.equals` when generating `equals()` methods, so I don't think we have do anything.